text/0000-tuple-struct-self-ctor.md

```diff
@@ -139,23 +224,116 @@ should work and will probably try at some point.
 # Reference-level explanation
 [reference-level-explanation]: #reference-level-explanation
 
-Rust (now) allows usage of `Self(v0, v1, ..)` inside inherent
-and trait `impl`s of tuple structs, either when mentioning the
-tuple struct directly in the `impl` header, or via a type alias.
+Rust (now) allows usage of `Self(v0, v1, ..)` inside inherent and
+trait `impl`s of tuple structs, either when mentioning the tuple struct
+directly in the `impl` header, or via a type alias.
+This usage is permitted both in expression and pattern contexts.
```

Comment:
I think I would prefer if we can rewrite this Reference-Level Explanation to be a little more "implementation focused". Right now it reads a bit to me like restating the premise of the RFC. For example, I think I would prefer to say:

When entering one of the following contexts, the value namespace is extended with `Self` which maps to the tuple constructor fn (or constant, in the case of a unit struct):

  - inherent impls where the `Self` type is a tuple or unit struct
  - trait impls where the `Self` type is a tuple or unit struct